includes/class-wc-stripe-blocks-support.php

```diff
@@ -60,6 +60,63 @@ public function is_active() {
 	 * @return array
 	 */
 	public function get_payment_method_script_handles() {
+		if ( WC_Stripe_Feature_Flags::is_upe_checkout_enabled() ) {
+			$this->register_upe_payment_method_script_handles();
+		} else {
+			$this->register_classic_payment_method_script_handles();
+		}
```

Comment:
I think this is a better approach to loading the relevant JS because it allows us to load smaller JS bundles.

The alternative would be to have the JS decide whether the UPE or "Classic" experience should be loaded, but that would mean a bigger JS bundle since the JS would contain both the "Classic" and UPE experiences; roughly half of the bundle would go unused in this case.

The downside of this approach is that the built plugin assets will be a bit bigger so merchants will need to download a slightly bigger plugin archive when installing/updating the plugin. I think the trade-off here is worth it though; a lighter (and thus faster) checkout experience is always worth it.

That said, I'm still open to suggestions on this 🙂